lib/utilities/async_button.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+
+class AsyncButton extends StatefulWidget {
+  final void Function() onPressed;
+  final String label;
+
+  const AsyncButton({required this.onPressed, required this.label, Key? key}) : super(key: key);
+
+  @override
+  State<AsyncButton> createState() => _AsyncButtonState();
+}
+
+class _AsyncButtonState extends State<AsyncButton> {
+  bool submitting = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton.icon(
+        label: Text(widget.label),
+        icon: !submitting
+            ? Container()
```

Comment:
🤔 For some reason using an empty `Container` didn't work for me in the context of [closing the channel](https://github.com/itchysats/10101/pull/394#discussion_r1033462062). We might need to use a `SizedBox.shrink()` instead?

Reply:
I had this problem when using the `ElevatedButton` (i.e. `AsyncButton`) as FAB, but I think we actually should not use it as FAB in our context, because what we have is more of a form submission rather than triggering an action.
I ran into this problem here as well: https://github.com/itchysats/10101/pull/398/commits/7a2b10645f283f5276c458abb9af0f67168cbbeb and fixed it by adding the Button to the body instead of using a FAB.

I am not sure this is 100% "correct", but I think the FAB makes sense when we have (one or multiple) action that apply to the screen. Once we are in a sub-screen where the user submits a form, I think it  makes more sense to integrate the button in the body as form submission.

Reply:
That makes sense, thanks for the explanation!